libs/metrics/test_positivity.py

```diff
@@ -155,10 +227,74 @@ def calculate(self, dataset: MultiRegionDataset, diff_days: int) -> MultiRegionD
         wide_date_df = df.loc[self._column, :]
         # Optional optimization: The following likely adds the variable/field/column name back in
         # to the index which was just taken out. Consider skipping reindexing.
-        return _make_output_dataset(dataset, self._name, wide_date_df, CommonFields.TEST_POSITIVITY)
+
+        return self._filter_recent(
+            _make_output_dataset(dataset, self._name, wide_date_df, CommonFields.TEST_POSITIVITY),
+            most_recent_date,
+        )
+
+
+class OldMethod(Method):
+    """A method of calculating test positivity using smoothed POSITIVE_TEST and NEGATIVE_TESTS."""
+
+    @property
+    def name(self) -> timeseries.DatasetName:
+        return timeseries.DatasetName("OldMethod")
+
+    @property
+    def columns(self) -> Set[FieldName]:
+        return {CommonFields.POSITIVE_TESTS, CommonFields.NEGATIVE_TESTS}
+
+    def calculate(
+        self, dataset: MultiRegionDataset, diff_days: int, most_recent_date
+    ) -> MethodOutput:
+
+        positivity_time_series = {}
+        # To replicate the behavior of the old code, a region is considered to have recent
+        # positivity when the input timeseries (POSITIVE_TESTS and NEGATIVE_TESTS) are recent. The
+        # other subclasses of `Method` filter based on the most recent real value in the output
+        # timeseries.
+        recent_region = set()
```

Comment:
`recent_regions`? `regions_with_recent_data`?  also, fwiw do we expect there to ever be overlaps in that for loop? I would guess that as we're grouping by region, it wouldn't happen and we could just use a list

Reply:
changed to list